src/simulations/Simulations.jl

```diff
@@ -198,14 +198,23 @@ function solve!(landsim::LandSimulation)
     SciMLBase.solve!(landsim._integrator)
 end
 
+"""
+    get_domain(model::AbstractModel)
+
+Return the ClimaLand domain of the model - for integrated models,
+this is the soil domain (e.g., Column, SphericalShell).
+"""
+get_domain(model::ClimaLand.AbstractModel) = model.domain
+get_domain(model::ClimaLand.AbstractLandModel) = model.soil.domain
+
 """
     ClimaComms.context(landsim::LandSimulation)
 
 Returns the context of the land simulation (distributed or
 on a single node).
 """
 function ClimaComms.context(landsim::LandSimulation)
-    return ClimaComms.context(landsim.model)
+    return ClimaComms.context(get_domain(landsim.model))
```

Comment:
I think this should have worked. We have this code:
```julia
function ClimaComms.context(model::AbstractModel)
    if :domain ∈ propertynames(model)
        return ClimaComms.context(model.domain)
    else
        error(
            "Your model does not contain a domain. If this is intended, you will need a new method of ClimaComms.context.",
        )
    end
end

function ClimaComms.device(model::AbstractModel)
    if :domain ∈ propertynames(model)
        return ClimaComms.device(model.domain)
    else
        error(
            "Your model does not contain a domain. If this is intended, you will need a new method of ClimaComms.device.",
        )
    end
end
```

Reply:
It doesnt work because for integrated models, the domain is not stored in model.domain (it's why I had to make the get_domain(model) function earlier)